src/metabase/lib/schema/aggregation.cljc

```diff
@@ -55,52 +84,54 @@
   #_expr       [:ref ::expression/number]
   #_percentile [:ref ::percentile.percentile])
 
-(lib.hierarchy/derive :percentile :lib.type-of/type-is-type-of-first-arg)
+(lib.hierarchy/derive :percentile ::aggregation)
+(lib.hierarchy/derive :percentile ::type-is-type-of-first-arg-or-number)
 
 (mbql-clause/define-tuple-mbql-clause :share :- :type/Float
   [:schema [:ref ::expression/boolean]])
 
+(lib.hierarchy/derive :share ::aggregation)
+
 (mbql-clause/define-tuple-mbql-clause :stddev :- :type/Float
   [:schema [:ref ::expression/number]])
 
+(lib.hierarchy/derive :stddev ::aggregation)
+
 (mbql-clause/define-tuple-mbql-clause :sum
   [:schema [:ref ::expression/number]])
 
+(lib.hierarchy/derive :sum ::aggregation)
+(lib.hierarchy/derive :sum ::type-is-type-of-first-arg-or-number)
+
 (mbql-clause/define-tuple-mbql-clause :cum-sum
   [:schema [:ref ::expression/number]])
 
-(lib.hierarchy/derive :sum :lib.type-of/type-is-type-of-first-arg)
-
-(lib.hierarchy/derive :cum-sum :lib.type-of/type-is-type-of-first-arg)
+(lib.hierarchy/derive :cum-sum ::aggregation)
+(lib.hierarchy/derive :cum-sum ::type-is-type-of-first-arg-or-number)
 
 (mbql-clause/define-tuple-mbql-clause :sum-where
   [:schema [:ref ::expression/number]]
   [:schema [:ref ::expression/boolean]])
 
-(lib.hierarchy/derive :sum-where :lib.type-of/type-is-type-of-first-arg)
+(lib.hierarchy/derive :sum-where ::aggregation)
+(lib.hierarchy/derive :sum-where ::type-is-type-of-first-arg-or-number)
 
 (mbql-clause/define-tuple-mbql-clause :var :- :type/Float
   #_expr [:schema [:ref ::expression/number]])
 
+(lib.hierarchy/derive :var ::aggregation)
+
 (mr/def ::aggregation
-  ;; placeholder!
-  [:or
-   :mbql.clause/avg
-   :mbql.clause/count
-   :mbql.clause/cum-count
-   :mbql.clause/count-where
-   :mbql.clause/distinct
-   :mbql.clause/max
-   :mbql.clause/median
-   :mbql.clause/min
-   :mbql.clause/percentile
-   :mbql.clause/share
-   :mbql.clause/stddev
-   :mbql.clause/sum
-   :mbql.clause/cum-sum
-   :mbql.clause/sum-where
-   :mbql.clause/var
-   any?])
+  [:and
+   ::mbql-clause/clause
+   [:or
+    ::expression/number
```

Comment:
This should be a numeric expression which contains an aggregation, right?

Reply:
I did think about this, and it turns out that in Postgres at least it's valid to do something like

```sql
SELCET count(*), 1 + 2
FROM my_table
GROUP BY 1 + 2
```

e.g. aggregate by a numeric expression that does not include any aggregate functions in it... 

It also seemed a little tricky/error-prone to try to write the schema to check that something was a numeric expression **and** contained at least one aggregation function, we'd probably have to do a `:fn` schema and use `mbql.u.match/match` or something to look for it. I'll add an issue to cleanup to do this later and I'll tackle that as a follow-on

Reply:
I opened https://github.com/metabase/metabase/issues/31914 to track this... I'll do it as a follow-on because I don't want to potentially break stuff in this PR by adding that restriction